contracts/mint/AnnualCappedInflation.sol

```diff
@@ -0,0 +1,48 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.19;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IDCNTMintAuthorization, IERC20 } from "./IDCNTMintAuthorization.sol";
+
+/// @notice an implementation of IDCNTMintAuthorization that limits mint requests to a capped annual inflation
+contract AnnualCappedInflation is Ownable, IDCNTMintAuthorization {
+    IERC20 public token;
+    uint128 public nextMint; // Timestamp
+    uint32 public constant MINIMUM_MINT_INTERVAL = 365 days;
+    uint8 public constant MINT_CAP_BPS = 200; // 2%
```

Comment:
Should this `MINT_CAP_BPS` be a non-constant value passed in through the constructor? To support if we decide that we want a capped annual inflation percentage other than 2%

Reply:
The jury is still out on what the best approach for Decent is regarding our "ability to mint after launch" strategy. Although I thought the approach of "allowing the DAO to upgrade the authorization logic, and then renounce the ability to  perform further upgrades" was sufficient, there are lots of arguments to bake the authorization in at launch.

With that being said, the `AnnualCappedInflation.sol` contract, which is simply an implementation of `IDCNTMintAuthorization`, exists in this codebase as an example.

_Anyone_ can deploy an instance of `IDCNTMintAuthorization` with whatever code they want, whenever they want. It's up to the DAO to pass a vote that performs the upgrade to this new instance on the DCNT token contract.

For sake of example, let's assume the DCNT token gets deployed along with the `NoMint.sol` implementation.  No instances of `AnnualCappedInflation.sol` are deployed yet. In 5 months, we determine that DCNT tokenomics do need inflation, but at the rate of 1% per year. In that case we would:

1. Change this constant in this `AnnualCappedInflation.sol` contract from `200` to `100`.
2. Someone would deploy it.
3. Someone would make an EVM proposal on the DAO to
  a. call `updateMintAuthorization` on DCNT with the address from step 2.
  b. call `renounceRole` on DCNT with the `UPDATE_MINT_AUTHORIZATION_ROLE`

Kind of a longwinded answer just to say "we'll only deploy these instances of `IDCNTMintAuthorization` as we need them, so I don't think it really matters to build them with too much dependency injection.